src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1294,6 +1294,37 @@ Node* ShiftVNode::Identity(PhaseGVN* phase) {
   return this;
 }
 
+Node* VectorMaskOpNode::Ideal(PhaseGVN* phase, bool can_reshape) {
```

Comment:
It doesn't make much sense to me. Why don't you simply require the input to be in canonical shape from the very beginning by unconditionally wrapping it into `VectorStoreMask` during construction?

Reply:
Done